### PR TITLE
- Firebase integration has been done and some events hav been set up.

### PR DESCRIPTION
**Pull request by yasirinitialcode**
## Related Issue
- Integration of Firebase.

ISSUE LINK: https://app.asana.com/0/1203144541549209/1204097456360678

## Description

- Firebase has been integrated into project.
- Some events were set up.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- Events were observed in Firebase console.

## Screenshots or Sample Output

<p float="left">
  

</p>

### CODE REVIEW

File: app/src/main/java/com/netmera/netmerafintech/ui/home/HomeFragment.kt [hunk 1]
```diff
@@ -23,9 +24,11 @@ class HomeFragment: Fragment() {
     private var card: Card = Card(CardType.BLUE.value)
 
     private val onManageClick = { card: Card ->
+        AnalyticsUtil.manageEvent()
         ManageCardActivity.start(requireActivity(), card)
     }
     private val onTransactionClick = { transaction: Transaction ->
+        transaction.transactionId?.let { AnalyticsUtil.paymentDetailEvent(it) }
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/ui/home/HomeFragment.kt]: You should do this check inside `AnalyticsUtil.paymentDetailEvent` IMO

File: app/src/main/java/com/netmera/netmerafintech/ui/home/ManageCardActivity.kt [hunk 2]
```diff
@@ -30,7 +33,11 @@ class ManageCardActivity: AppCompatActivity() {
         binding = ActivityManageCardBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        val card: Card? = intent.extras?.parcelable(ARG_CARD)
+        val bundle: Bundle? = intent.extras
+        bundle?.let {
+            it.classLoader = ManageCardActivity::class.java.classLoader
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/ui/home/ManageCardActivity.kt:39]: Why did you add `classLoader`?

File: app/src/main/java/com/netmera/netmerafintech/ui/payments/SendMoneyActivity.kt [hunk 3]
```diff
@@ -75,7 +82,14 @@ class SendMoneyActivity : AppCompatActivity() {
         binding.apply {
             backButton.setOnClickListener { finish() }
             changeButton.setOnClickListener{ toast("Change event was called") }
-            sendButton.setOnClickListener { toast("Send money event was called") }
+            sendButton.setOnClickListener {
+                if (amount.text.toString() != "" || amount.text.toString() != "0.00") {
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/ui/payments/SendMoneyActivity.kt]: Shouldn't it be `AND`?
```suggestion
                if (amount.text.toString() != "" && amount.text.toString() != "0.00") {
```

File: app/src/main/java/com/netmera/netmerafintech/utils/AnalyticUtils.kt [hunk 4]
```diff
@@ -1,31 +1,170 @@
...
-        val netmerauser = NMImpactFintechUser()
-        netmerauser.setUserId("burakaymaz@hotmail.com")
-        netmerauser.setName("Burak")
-        netmerauser.setType("black")
-        Netmera.updateUser(netmerauser)
+    //region MANAGE EVENT
+    fun manageEvent() {
+        manageNetmeraEvent()
+        manageFirebaseEvent()
+    }
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/utils/AnalyticUtils.kt]: Can you divide this file as `NetmeraAnalyticsUtil` and `FirebaseAnalyticsUtil`? Then you can call it like
```suggestion
    fun manageEvent() {
        NetmeraAnalyticsUtil.manageEvent()
        FirebaseAnalyticsUtil.manageEvent()
    }
```

Comment (ghost) [reply app/src/main/java/com/netmera/netmerafintech/utils/AnalyticUtils.kt]: That would be more organized.

**Review by ghost — COMMENTED** [review]
Thank you, but small changes needed

File: app/src/main/java/com/netmera/netmerafintech/ui/payments/SendMoneyActivity.kt (hunk 3 above)
Comment (yasirinitialcode) [reply app/src/main/java/com/netmera/netmerafintech/ui/payments/SendMoneyActivity.kt]: You're right.

File: app/src/main/java/com/netmera/netmerafintech/ui/home/ManageCardActivity.kt (hunk 2 above)
Comment (yasirinitialcode) [reply app/src/main/java/com/netmera/netmerafintech/ui/home/ManageCardActivity.kt:39]: There was a issue called `Class not found when unmarshalling`. That is the recommended solution.

Comment (ghost) [reply app/src/main/java/com/netmera/netmerafintech/ui/home/ManageCardActivity.kt:39]: Can you share the issue and recommended solution here. I would like to take a look

Comment (yasirinitialcode) [reply app/src/main/java/com/netmera/netmerafintech/ui/home/ManageCardActivity.kt:39]: https://stackoverflow.com/a/28589962

File: app/src/main/java/com/netmera/netmerafintech/utils/IAnalyticsUtil.kt [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/*
+* Copyright (c) 2023 Netmera.
+*/
+
+package com.netmera.netmerafintech.utils
+
+import com.netmera.netmerafintech.data.model.Card
+
+interface IAnalyticsUtil {
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/utils/IAnalyticsUtil.kt:10]: Very good.

**Review by ghost — APPROVED** [review]
Thank you

File: app/src/main/java/com/netmera/netmerafintech/utils/AnalyticsUtil.kt [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        FirebaseAnalyticsUtil.signInEvent()
+    }
+
+    override fun signOutEvent() {
+        NetmeraAnalyticsUtil.signOutEvent()
+        FirebaseAnalyticsUtil.signOutEvent()
+    }
+
+    fun updateUser() {
+        NetmeraAnalyticsUtil.updateUser()
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/AnalyticsUtil.kt]: You can also set userId on the firebase side.

**Review by bsoykal — APPROVED** [review]
Good job, but please also add setUserId on the Firebase side.

File: app/src/main/java/com/netmera/netmerafintech/utils/AnalyticsUtil.kt [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        NetmeraAnalyticsUtil.signInEvent()
+        FirebaseAnalyticsUtil.signInEvent()
+    }
+
+    override fun signOutEvent() {
+        NetmeraAnalyticsUtil.signOutEvent()
+        FirebaseAnalyticsUtil.signOutEvent()
+    }
+
+    override fun userUpdate() {
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/AnalyticsUtil.kt]: We should create a user class and pass all attributes as a method parameter.

**Review by bsoykal — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/netmera/netmerafintech/ui/splash/SplashActivity.kt [hunk 8]
```diff
@@ -33,7 +35,10 @@ class SplashActivity : AppCompatActivity() {
                 startActivity(Intent(this@SplashActivity, AllPagesActivity::class.java))
             }
             signInButton.setOnClickListener {
-                AnalyticsUtil.userUpdate()
+                user.userName = "Burak"
+                user.userId = "burakaymaz@hotmail.com"
+                user.type = "black"
+                AnalyticsUtil.userUpdate(user)
                 AnalyticsUtil.signInEvent()
                 toast("User update was done and sign in event and was called")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/ui/splash/SplashActivity.kt]: The user update is completed and the sign-In event was sent.

**Review by bsoykal — APPROVED** [review]
Please adjust the toast message, which I have mentioned. Thank you

**Pull request merged**